portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -135,9 +200,15 @@ async function placeProximityPins() {
   const pins = await response.json();
   const radius = Number(document.getElementById('radius').value);
 
-  pins.forEach(async (pin) => {
+  pins.forEach(async (pin, index, array) => {
     if (await haversineDistance(userAddress, pin.address) < radius) {
-      codeAddress(pin.address);
+      const distance = await haversineDistance(userAddress, pin.address);
+      if (index == array.length - 1) {
```

Comment:
Hm, why are we centering the graph on the last pin? Arbitrary decision? Why not userAddress?

That aside, if we're running addProximityPinAndWindow() in both blocks of the if statement you probably want to restructure it as:

```javascript
addProximityPinAndWindow(pin, distance);
if (index == array.length - 1){
  map.setCenter(addressToCoordinates(pin.address)[2]);
}

```

Reply:
arbitrary honestly, I think it's a better idea to center it on the user.